File: src/test/java/com/ohgnarly/fileripper/factories/FileServiceFactoryTest.java
package com.ohgnarly.fileripper.factories;

import com.ohgnarly.fileripper.models.FileDefinition;
import com.ohgnarly.fileripper.services.FileService;
import com.ohgnarly.fileripper.services.FlatFileService;
import com.ohgnarly.fileripper.services.XmlFileService;
import org.junit.Before;
import org.junit.Test;

import static com.ohgnarly.fileripper.integration.utilities.DataUtility.*;
import static org.junit.Assert.assertTrue;

public class FileServiceFactoryTest {
    private FileServiceFactory fileServiceFactory;

    @Before
    public void setUp() {
        fileServiceFactory = new FileServiceFactoryImpl();
    }

    @Test
    public void testCreateFileService_GivenDelimitedFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        FileDefinition fileDefinition = createDelimitedFileDefinition(",");

        //act
        FileService fileService = fileServiceFactory.createFileService(fileDefinition);

        //assert
        assertTrue(fileService instanceof FlatFileService);
    }

    @Test
    public void testCreateFileService_GivenFixedFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        FileDefinition fileDefinition = createFixedFileDefinition();

        //act
        FileService fileService = fileServiceFactory.createFileService(fileDefinition);

        //assert
        assertTrue(fileService instanceof FlatFileService);
    }

    @Test
    public void testCreateFileService_GivenXmlFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        FileDefinition fileDefinition = createXmlFileDefinition();

        //act
        FileService fileService = fileServiceFactory.createFileService(fileDefinition);

        //assert
        assertTrue(fileService instanceof XmlFileService);
    }
}

File: src/test/java/com/ohgnarly/fileripper/factories/FileServiceFactoryTest.kt
package com.ohgnarly.fileripper.factories

import com.ohgnarly.fileripper.services.FlatFileService
import com.ohgnarly.fileripper.services.XmlFileService
import com.ohgnarly.fileripper.testhelpers.buildDelimitedFileDefinition
import com.ohgnarly.fileripper.testhelpers.buildFixedFileDefinition
import com.ohgnarly.fileripper.testhelpers.buildXmlFileDefinition
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class FileServiceFactoryTest {
    private var fileServiceFactory: FileServiceFactory? = null

    @Before
    fun setUp() {
        fileServiceFactory = DefaultFileServiceFactory()
    }

    @Test
    fun testCreateFileService_GivenDelimitedFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        val fileDefinition = buildDelimitedFileDefinition(",")

        //act
        val fileService = fileServiceFactory!!.createFileService(fileDefinition)

        //assert
        Assert.assertTrue(fileService is FlatFileService)
    }

    @Test
    fun testCreateFileService_GivenFixedFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        val fileDefinition = buildFixedFileDefinition()

        //act
        val fileService = fileServiceFactory!!.createFileService(fileDefinition)

        //assert
        Assert.assertTrue(fileService is FlatFileService)
    }

    @Test
    fun testCreateFileService_GivenXmlFileDefinition_ShouldReturnDelimitedFileService() {
        //arrange
        val fileDefinition = buildXmlFileDefinition()

        //act
        val fileService = fileServiceFactory!!.createFileService(fileDefinition)

        //assert
        Assert.assertTrue(fileService is XmlFileService)
    }
}